### ISSUE TITLE
with recursive clause not supported

### ISSUE THREAD

**Issue by qiqingli** [nene/sql-parser-cst#100]
```
with recursive 
standard_address as (
        select id, parent_id,
            grade
        from pb_standard_addr
        where
            id = :addressId::numeric
        union
        select e.id, e.parent_id,
            e.grade
        from
            pb_standard_addr e
        inner join standard_address s on s.parent_id = e.id
    )
select * from standard_address;
```

```
["ERROR" - 4:50:23 PM] Error formatting document.
["ERROR" - 4:50:23 PM] Syntax Error: Unexpected "standard_address"
Was expecting to see: "(", "AS", or whitespace
  |
2 | standard_address as (
  | ^
Error: Syntax Error: Unexpected "standard_address"
Was expecting to see: "(", "AS", or whitespace
  |
2 | standard_address as (
  | ^
    at parse (/Users/qiqingli/node_modules/sql-parser-cst/lib/main.js:40:19)
    at Object.parse (/Users/qiqingli/node_modules/prettier-plugin-sql-cst/dist/index.js:41:76)
    at parse5 (file:///Users/qiqingli/node_modules/prettier/index.mjs:19838:24)
    at async coreFormat (file:///Users/qiqingli/node_modules/prettier/index.mjs:20386:25)
    at async formatWithCursor (file:///Users/qiqingli/node_modules/prettier/index.mjs:20598:14)
    at async Module.format2 (file:///Users/qiqingli/node_modules/prettier/index.mjs:22002:25)
["INFO" - 4:50:23 PM] Formatting completed in 328ms.
```

**Comment by nene** [comment]
Thanks for reporting.

Looks like some keywords are mistakenly not treated as case-insensitive. The above works fine when using uppercase `RECURSIVE`, but not with lowercase `recursive`.

**Comment by nene** [comment]
Fixed in 0.31.1
